src/axom/core/Array.hpp

```diff
@@ -728,48 +837,65 @@ class Array
   }
   /// @}
 
-  T* m_data;
-  IndexType m_num_elements;
-  IndexType m_capacity;
-  double m_resize_ratio;
-  bool m_is_external;
+  T* m_data = nullptr;
+  /// \brief The full number of elements in the array
+  ///  i.e., 3 for a 1D Array of size 3, 9 for a 3x3 2D array, etc
+  IndexType m_num_elements = 0;
+  IndexType m_capacity = 0;
+  double m_resize_ratio = DEFAULT_RESIZE_RATIO;
+  bool m_is_external = false;
   int m_allocator_id;
+  // FIXME: What's the best way to make these available to users
```

Comment:
Following your comments about, we should try to expose this similarly to how numpy does

Reply:
numpy uses public attributes (`arr.shape` and `arr.strides`), which is fine for Python but maybe a little strange for C++, especially if we want to allow modification (reshaping, in numpy terms).

Maybe it would make sense to instead use `std::array` in place of the raw arrays and return const refs to that?

Reply:
moving to `std:;array` for those would also clean up some of the other implementation details (could just use `std::array` constructors rather than `std::copy`ing contents around)